### DISCUSSION TITLE
Pagination SEO issue: canonical url on paginated pages does not work

### DISCUSSION THREAD

**Discussion by krzysiekpaszek** [thuliteio/doks#344] [Q&A]
On pages with numbers: ../page/2/ ...
it shows the canonical for the main taxonomy page, without the page number this is a problem for indexing.
As described here:
https://www.searchenginejournal.com/technical-seo/pagination/

this is a general issue with hugo described here:
https://github.com/gohugoio/hugo/issues/4507

In DOKS pagination and canonicalization is done in seo.html in partials/head

However, this:
`{{ if $.Scratch.Get "paginator" }}
  <link rel="canonical" href="{{ .Paginator.URL | absLangURL }}page/{{ .Paginator.PageNumber }}/">
  {{ if .Paginator.HasPrev -}}
    <link rel="prev" href="{{ .Paginator.Prev.URL | absLangURL }}">
  {{ end -}}
  {{ if .Paginator.HasNext -}}
    <link rel="next" href="{{ .Paginator.Next.URL | absLangURL }}">
  {{ end -}}
{{ else -}}
  <link rel="canonical" href="{{ .Permalink }}">
{{ end -}}`
does not work
on pagination pages the ELSE is executed, and no prev/next and canonical is done

way around the Hugo problem with canonical for pagination pages is described here:
https://moonbooth.com/hugo/seo/#canonicals

**Comment by h-enk** [comment]
Hi @krispkrisp, thanks for your input! Am looking into it + working on it.  Will keep you posted.

**Reply by h-enk** [reply]
@krispkrisp, could you verify if the problem still exists w/ Doks [v0.3.0](https://github.com/h-enk/doks/releases/tag/v0.3.0)? Thanks!

**Reply by krzysiekpaszek** [reply]
@h-enk not working, same thing
I have created new install, with dependencies, I have just copied the folder with about 100 files
the pagintion was build, however on pagination pages ../page/2/, page/3/ ...
the canonical points to the main section, and not to specific page.

**Reply by h-enk** [reply]
@krispkrisp, thanks for checking. Will be looking into it.

**Reply by h-enk** [reply]
#359

**Comment by h-enk** [comment]
@krispkrisp, fixed w/ a1dce26c2450359e02743ac6289f53179971e929

Added paginator and  `{{ $.Scratch.Set "paginator" true }}` to `./layouts/blog/list.html`:

```html
..
{{ $paginator := .Paginate (.Data.Pages) -}}
{{ range $paginator.Pages -}}
  ..
{{ end -}}
{{ $.Scratch.Set "paginator" true }}
{{ template "_internal/pagination.html" . }}
..
```

And there you go:

![Snag_246ec9c6](https://user-images.githubusercontent.com/3902872/132646456-b3df34fd-c7b6-4335-aa4f-5d66b91740da.png)
